### PR TITLE
reshape on NDArray not working?

### PR DESCRIPTION
**Pull request by iblis17**
```julia
julia> x = mx.zeros(3, 4)

julia> y = mx.reshape(x, shape=(4, 3)) 
mx.NDArray{Float32}(3, 4)         

julia> size(y)
(3, 4)

julia> copy(y) 
3×4 Array{Float32,2}:   
 0.0  0.0  0.0  0.0 
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
```

### CODE REVIEW

**Comment by vchuravy** [comment]
This might be also necessary for the Symbolic API?

**Comment by iblis17** [comment]
ha, I do not check Sumbolic API yet. 
I'm reading [c_api.h](https://github.com/apache/incubator-mxnet/blob/470b56437290b33fef51b067c96fdce08cefa584/include/mxnet/c_api.h) line-by-line :p 
And still no idea how Symbolic related stuff works.

File: src/ndarray.jl [hunk 1]
```diff
@@ -921,6 +921,25 @@ function save(filename::String, data::Dict{Base.Symbol,NDArray})
...
+import Base: reshape
+
+reshape{N}(arr::NDArray, dim::NTuple{N, Integer}) = _reshape(arr, dim)
+reshape{N}(arr::NDArray, dim::Vararg{Integer, N}) = _reshape(arr, dim)
+
+@inline function _reshape{N}(arr::NDArray, dim::NTuple{N, Integer})
+  h_ref = Ref{MX_handle}(C_NULL)
+  cint_dim = reverse(collect(Cint, dim))
+  @mxcall(:MXNDArrayReshape, (MX_handle, Cint, Ref{Cint}, Ref{MX_handle}),
+          arr, N, Ref(cint_dim), h_ref)
```
Comment (iblis17) [inline src/ndarray.jl]: I also found that there are a `reshape` operator support NDArray reshaping. what's the main difference? And... seems that (before this PR) we are using the `reshape` operator on both NDArray and SymbolicNode.

Comment (vchuravy) [reply src/ndarray.jl]: Looking at the two implementation the `reshape` operator seems to be more flexible and we should probably use them.

**Comment by iblis17** [comment]
switched to operator already.
ready for review

**Comment by pluskid** [comment]
I agree the difference between row-based indexing and column based indexing in Julia and C++ backend are causing some gray areas. Especially about the meaning of the arguments for operators like reshape, transpose / swapaxes, etc. This issue is made worse with the auto-generated doc from the C++ backend which usually focus on describing the Python behavior.

**Pull request merged**